### PR TITLE
Stitcher Add new extractor (Issue: #7162)

### PR DESCRIPTION
**Pull request by mjdubell**
New extractor

### CODE REVIEW

File: youtube_dl/extractor/stitcher.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        'md5': '391dd4e021e6edeb7b8e68fbf2e9e940',
+        'info_dict': {
+            'id': '40789481',
+            'ext': 'mp3',
+            'title': 'Machine Learning Mastery and Cancer Clusters from Talking Machines',
+        }
+    }
+
+    def _real_extract(self, url):
+        audio_id = self._search_regex(r'[a-z\/\-\:\/\/.]+?(\d+?)\?.+', url, "audio_id")
```
Comment (jaimeMF) [inline youtube_dl/extractor/stitcher.py]: The method we follow here is to use the `_VALID_URL`, which you would define as:
```python
_VALID_URL = r'https?://(?:www\.)?stitcher\.com/podcast/[\/a-z\-]+(?P<id>\d+)\?.+'
```
And then you only need to do:
```python
audio_id = self._match_id(url)
```

File: youtube_dl/extractor/stitcher.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        webpage = self._download_webpage(url, audio_id)
+
+        title = self._og_search_title(webpage)
+        url = self._search_regex(r'[\s\S]*episodeURL: "(.+?)"[\s\S]*', webpage, 'url')
```
Comment (dstftw) [inline youtube_dl/extractor/stitcher.py]: `[\s\S]*` on both sides are noop due to `*`.

File: youtube_dl/extractor/stitcher.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        episode_image = self._search_regex(r'[\s\S]*episodeImage: "(.+?)"[\s\S]*', webpage, 'thumbnail')
```
Comment (dstftw) [inline youtube_dl/extractor/stitcher.py]: Should be `fatal=False`.

File: youtube_dl/extractor/stitcher.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        duration = int(self._search_regex(r'[\s\S]*duration: (\d+?),[\s\S]*', webpage, 'duration')) / 60
```
Comment (dstftw) [inline youtube_dl/extractor/stitcher.py]: Should be `fatal=False`. Use `utils.int_or_none`. It must be in seconds.

**Comment by mjdubell** [comment]
Updated the code

File: youtube_dl/extractor/stitcher.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# coding: utf-8
+from __future__ import unicode_literals
+from .common import InfoExtractor
+
+
+class StitcherIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?stitcher\.com/podcast/[\/a-z\-]+(?P<id>\d+)|\?[a-z=]+'
```
Comment (dstftw) [inline youtube_dl/extractor/stitcher.py]: What are you trying to achieve with `|\?[a-z=]+`?

File: youtube_dl/extractor/stitcher.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    def _real_extract(self, url):
+        audio_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, audio_id)
+
+        title = self._og_search_title(webpage)
+        url = self._search_regex(r'episodeURL: "(.+?)"', webpage, 'url')
+        episode_image = self._search_regex(r'episodeImage: "(.+?)"', webpage, 'episode_image', fatal=False)
+        duration = self._search_regex(r'simpleDuration: "(\d+?) minutes"', webpage, 'duration', fatal=False)
```
Comment (dstftw) [inline youtube_dl/extractor/stitcher.py]: As said duration must be int.

File: youtube_dl/extractor/stitcher.py (hunk 5 above)
Comment (mjdubell) [inline youtube_dl/extractor/stitcher.py]: I was trying to match:
```
?autoplay=true
```
But I see now that isn't necessary.

**Comment by mjdubell** [comment]
Updated code

**Pull request closed without merging**